### PR TITLE
fix: Reuse mmkv instance once created

### PR DESCRIPTION
**Pull request by tommasini**
## **Description**
We should reuse MMKV instance instead of creating a new one each time we want to access storage.

## **Related issues**

N/A

## **Manual testing steps**

1. Fresh install the app
2. Press import with SRP
3. Accept share data and terms of use
4. kill the app
5. Open it
6. go to import with SRP again
7. Should not show the same modals (This means it's being saved and read under MMKV)

## **Screenshots/Recordings**

https://github.com/user-attachments/assets/2d342876-9675-482c-8d54-f6b4e1258100

### **Before**

N/A

### **After**

N/A

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/store/storage-wrapper.js [hunk 1]
```diff
@@ -66,4 +73,4 @@ class StorageWrapper {
   }
 }
 
-export default new StorageWrapper();
+export default StorageWrapper.instance || new StorageWrapper();
```
[the remark this answers is not in the document]
Comment (tommasini) [reply app/store/storage-wrapper.js]: Thanks Nico for the meeting and helping get more perfect towards a Singleton pattern design!

File: app/store/storage-wrapper.test.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
+jest.unmock('./storage-wrapper');
```
Comment (NicolasMassart) [inline app/store/storage-wrapper.test.ts:1]: S: you could add a small comment here to explain why you unmock.
```suggestion
// Unmocking storage-wrapper as it's mocked in testSetup directly
// to allow easy testing of other parts of the app
// but here we want to actually test storage-wrapper
jest.unmock('./storage-wrapper');
```

File: app/store/storage-wrapper.test.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    const getItemSpy = jest.spyOn(StorageWrapper, 'getItem');
+
+    await StorageWrapper.setItem('test-key', 'test-value');
+    getItemSpy.mockResolvedValue('test-value');
+
+    const result = await StorageWrapper.getItem('test-key');
+
+    expect(setItemSpy).toHaveBeenCalledWith('test-key', 'test-value');
+    expect(getItemSpy).toHaveBeenCalledWith('test-key');
+    expect(result).toBe('test-value');
```
Comment (NicolasMassart) [inline app/store/storage-wrapper.test.ts:16]: C: not sure this one actually tests anything except mockResolvedValue...

File: app/store/storage-wrapper.test.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  });
+  it('throw an error when setItem do not receives a string', async () => {
```
Comment (NicolasMassart) [inline app/store/storage-wrapper.test.ts:18]:
```suggestion
  it('throws when setItem value param is not a string', async () => {
```

File: app/store/storage-wrapper.test.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+      expect(e).toBeInstanceOf(Error);
+      expect(e.message).toBe(
+        'MMKV value must be a string, received value 123 for key test-key',
+      );
+    }
+
+    expect(setItemSpy).toHaveBeenCalledWith('test-key', 123);
+  });
+
+  it('remove value from the store', async () => {
```
Comment (NicolasMassart) [inline app/store/storage-wrapper.test.ts:35]:
```suggestion
  it('removes value from the store', async () => {
```

File: app/store/storage-wrapper.test.ts [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    const removeItemSpy = jest.spyOn(StorageWrapper, 'removeItem');
+    await StorageWrapper.setItem('test-key', 'test-value');
+    const resultBeforeRemove = await StorageWrapper.getItem('test-key');
+    expect(resultBeforeRemove).toBe('test-value');
```
Comment (NicolasMassart) [inline app/store/storage-wrapper.test.ts:39]: S: Add some empty line to make test steps separated and more readable
```suggestion
    expect(resultBeforeRemove).toBe('test-value');

```

File: app/store/storage-wrapper.test.ts [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    await StorageWrapper.removeItem('test-key');
+    expect(removeItemSpy).toHaveBeenCalledWith('test-key');
```
Comment (NicolasMassart) [inline app/store/storage-wrapper.test.ts:41]: S: Add some empty line to make test steps separated and more readable
```suggestion
    expect(removeItemSpy).toHaveBeenCalledWith('test-key');

```

File: app/store/storage-wrapper.test.ts [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    const resultAfterRemoval = await StorageWrapper.getItem('test-key');
+    expect(resultAfterRemoval).toBeNull();
+  });
+
+  it('remove all values from the store', async () => {
```
Comment (NicolasMassart) [inline app/store/storage-wrapper.test.ts:46]:
```suggestion
  it('removes all values from the store', async () => {
```

File: app/store/storage-wrapper.test.ts [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+    await StorageWrapper.clearAll();
+    expect(clearAllSpy).toHaveBeenCalled();
+    const result = await StorageWrapper.getItem('test-key');
+    const result2 = await StorageWrapper.getItem('test-key-2');
+    expect(result).toBeNull();
+    expect(result2).toBeNull();
+  });
+
+  it('StorageWrapper instance is defined', () => {
```
Comment (NicolasMassart) [inline app/store/storage-wrapper.test.ts:63]: S: The suite already is called "StorageWrapper", no need to repeat.
```suggestion
  it('singleton instance is defined', () => {
```
Then your test will appear as "StorageWrapper singleton instance is defined"

File: app/store/storage-wrapper.ts [hunk 10]
```diff
@@ -4,26 +4,21 @@ import { isE2E } from '../util/test/utils';
...
+  private constructor() {
+    /**
+     * The underlying storage implementation.
+     * Use `ReadOnlyNetworkStore` in test mode otherwise use `AsyncStorage`.
+     */
+    this.storage = isE2E ? ReadOnlyNetworkStore : new MMKV();
   }
 
-  async getItem(key) {
+  async getItem(key: string) {
```
Comment (NicolasMassart) [inline app/store/storage-wrapper.ts:21]: S: add some TSDoc to make IDE display it when we code.
Don't hesitate to add an `@example` section

File: app/store/storage-wrapper.ts [hunk 11]
```diff
@@ -40,7 +35,7 @@ class StorageWrapper {
     }
   }
 
-  async setItem(key, value) {
+  async setItem(key: string, value: string) {
```
Comment (NicolasMassart) [inline app/store/storage-wrapper.ts:38]: S: add some TSDoc to make IDE display it when we code.
Don't hesitate to add an `@example` section

File: app/store/storage-wrapper.ts [hunk 12]
```diff
@@ -56,7 +51,7 @@ class StorageWrapper {
     }
   }
 
-  async removeItem(key) {
+  async removeItem(key: string) {
```
Comment (NicolasMassart) [inline app/store/storage-wrapper.ts:54]: S: add some TSDoc to make IDE display it when we code.
Don't hesitate to add an `@example` section

File: app/store/storage-wrapper.ts [hunk 13]
```diff
@@ -71,6 +66,13 @@ class StorageWrapper {
   async clearAll() {
```
Comment (NicolasMassart) [inline app/store/storage-wrapper.ts:66]: S: add some TSDoc to make IDE display it when we code.
Don't hesitate to add an `@example` section

File: app/store/storage-wrapper.ts [hunk 14]
```diff
@@ -71,6 +66,13 @@ class StorageWrapper {
...
     await this.storage.clearAll();
   }
+
+  static getInstance() {
```
Comment (NicolasMassart) [inline app/store/storage-wrapper.ts:70]: S: add some TSDoc to make IDE display it when we code.
Don't hesitate to add an `@example` section

**Review by NicolasMassart — CHANGES_REQUESTED** [review]

File: app/store/storage-wrapper.test.ts (hunk 2 above)
Comment (tommasini) [reply app/store/storage-wrapper.test.ts:1]: Thanks Nico! Yeah sounds great to explain why that was needed!

File: app/store/storage-wrapper.test.ts (hunk 3 above)
Comment (tommasini) [reply app/store/storage-wrapper.test.ts:16]: Yeah, I will remove the mock resolved value, is not needed indeed!

**Review by NicolasMassart — APPROVED** [review]

**Pull request merged**